### PR TITLE
feat: simplify tests

### PR DESCRIPTION
**Pull request by olenagerasimova**
part of #1099 
I've simplified tests by extending `RestApiServerBase#requestAndAssert` method functionality and using it to avoid duplications.

### CODE REVIEW

File: src/test/java/com/artipie/api/RestApiServerBase.java [hunk 1]
```diff
@@ -124,9 +127,10 @@ final void beforeEach(final Vertx vertx, final VertxTestContext context) throws
      */
     final void requestAndAssert(final Vertx vertx, final VertxTestContext ctx,
         final Request rqs, final Consumer<HttpResponse<Buffer>> assertion) throws Exception {
-        WebClient.create(vertx)
-            .request(rqs.method, this.port(), RestApiServerBase.HOST, rqs.path)
-            .send()
+        final HttpRequest<Buffer> request = WebClient.create(vertx)
+            .request(rqs.method, this.port(), RestApiServerBase.HOST, rqs.path);
+        rqs.body.map(item -> request.sendJsonObject(rqs.body.get()))
```
Comment (dgarus) [inline src/test/java/com/artipie/api/RestApiServerBase.java:132]: `item` and  `rqs.body.get()` is the same, isn't it?

File: src/test/java/com/artipie/api/RestApiServerBase.java [hunk 2]
```diff
@@ -190,14 +194,40 @@ static final class Request {
...
+        private Optional<JsonObject> body;
+
         /**
          * Ctor.
          * @param method Http method
          * @param path Request path
+         * @param body Request body
          */
-        Request(final HttpMethod method, final String path) {
+        Request(final HttpMethod method, final String path, final Optional<JsonObject> body) {
```
Comment (dgarus) [inline src/test/java/com/artipie/api/RestApiServerBase.java:208]: To avoid misleading, let's rename `Request` to `TestRequest`.

**Review by dgarus — CHANGES_REQUESTED** [review]
@olenagerasimova thanks!
There are minor comments.

**Comment by olenagerasimova** [comment]
@dgarus thanks, corrected

**Review by dgarus — APPROVED** [review]
@olenagerasimova thanks!
LGTM

**Pull request merged**